ui/partials/perf/graphsctrl.html

```diff
@@ -24,10 +27,10 @@
         </tr>
       </table>
       <button class="btn btn-primary" ng-click="addTestData()">
-        <span class="glyphicon glyphicon-plus" aria-hidden="true"></span> Add more test data</button>
+        <span class="glyphicon glyphicon-plus" aria-hidden="true"></span> Add <span ng-show="seriesList.length > 0">more</span> test data</button>
     </div>
     <div id="data-display">
-      <select ng-model="myTimerange" ng-options="timerange.text for timerange in timeranges track by timerange.value" ng-change="timeRangeChanged()">
+      <select ng-model="myTimerange" ng-options="timerange.text for timerange in timeranges track by timerange.value" ng-change="timeRangeChanged() " ng-disabled="seriesList.length == 0">
```

Comment:
Looks like you added an extra space after timeRangeChanged() which we don't want --> ng-change="timeRangeChanged() "